src/containers/BooksForm.js

```diff
@@ -1,32 +1,94 @@
-import React from 'react';
-
-const BooksForm = () => {
-  const categories = [
-    'Action',
-    'Biography',
-    'History',
-    'Horror',
-    'Kids',
-    'Learning',
-    'Sci-Fi',
-  ];
-
-  return (
-    <form>
-      <input type="text" placeholder="Title" />
-      <select>
-        <option disabled>
-          Select Category
-        </option>
-        {categories.map(category => (
-          <option value={category} key={category}>
-            {category}
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import actions from '../actions/index';
+
+const { createBook } = actions;
+
+class BooksForm extends Component {
+  constructor(props) {
+    super(props);
+
+    this.createBook = actions;
+
+    this.state = {
+      title: '',
+      category: '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.reset = this.reset.bind(this);
+  }
+
+  handleChange(e) {
+    const { name, value } = e.target;
+
+    this.setState({
+      [name]: value,
+    });
+  }
```

Comment:
according to your state
```
    this.state = {
       title: '',
       category: '',
    };
```
State is an option with 2 properties, why's your `setState` having just to set one of the properties ??

Reply:
Hello Shey, 

For us to apply the DRY rule, we did it that way so it will track the changes for the title and the category fields (input field and select field respectively), they each have a name (title or category), and we will only update the state that has been changed at the time. Still, both of the states are going to be updated.